### PR TITLE
Feature lwdev 5567 extracting client api

### PR DESCRIPTION
**Pull request by dmitrysmall**
PR to TEST after QA verification on DEV.

### CODE REVIEW

File: src/Lykke.Service.CandlesHistory.Core/Domain/Candles/MarketType.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+﻿namespace Lykke.Service.CandlesHistory.Core.Domain.Candles
+{
+    public enum MarketType
+    {
+        Spot,
+        Mt
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.CandlesHistory.Core/Domain/Candles/MarketType.cs]: Give a full name

File: src/Lykke.Service.CandlesHistory.Services/Candles/InMemoryCandlesCacheService.cs [hunk 2]
```diff
@@ -8,10 +8,11 @@
 using Lykke.Job.CandlesProducer.Contract;
 using Lykke.Service.CandlesHistory.Core.Domain.Candles;
 using Lykke.Service.CandlesHistory.Core.Services.Candles;
+using System.Threading.Tasks;
 
 namespace Lykke.Service.CandlesHistory.Services.Candles
 {
-    public class CandlesCacheService : ICandlesCacheService
+    public class InMemoryCandlesCacheService : ICandlesCacheService
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.CandlesHistory.Services/Candles/InMemoryCandlesCacheService.cs]: Can't find where this class is used

File: src/Lykke.Service.CandlesHistory.Services/Candles/RedisCachedCandle.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using MessagePack;
+
+namespace Lykke.Service.CandlesHistory.Services.Candles
+{
+    [MessagePackObject]
+    public class RedisCachedCandle
+    {
+        [Key(1)]
+        public double Open { get; set; }
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.CandlesHistory.Services/Candles/RedisCachedCandle.cs]: Why not decimal? For all properties

File: src/Lykke.Service.CandlesHistory.Services/Candles/RedisCandlesCacheService.cs [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+                }
+            }
+
+            // TODO: This is non concurrent-safe update
+
+            var key = GetKey(assetPairId, priceType, timeInterval);
+
+            // Cleans cache
+
+            await _database.SortedSetRemoveRangeByRankAsync(key, 0, -1);
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.CandlesHistory.Services/Candles/RedisCandlesCacheService.cs]: Why not just delete the key?

File: src/Lykke.Service.CandlesHistory.Services/Candles/RedisCandlesCacheService.cs [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+
+            foreach (var candlesBatch in candles.Batch(100))
+            {
+                var entites = candlesBatch
+                    .Select(candle => new SortedSetEntry(SerializeCandle(candle), 0))
+                    .ToArray();
+
+                await _database.SortedSetAddAsync(key, entites);
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.CandlesHistory.Services/Candles/RedisCandlesCacheService.cs]: Use redit.Batch. The example https://github.com/LykkeCity/Lykke.Service.FixGateway/blob/dev/src/Lykke.Service.FixGateway.Services/ClientOrderIdProvider.cs#L45

Comment (MikhailGerasimov) [reply src/Lykke.Service.CandlesHistory.Services/Candles/RedisCandlesCacheService.cs]: I would add some reasonable TTL

File: src/Lykke.Service.CandlesHistory/DependencyInjection/ApiModule.cs [hunk 6]
```diff
@@ -65,12 +62,24 @@ protected override void Load(ContainerBuilder builder)
...
+
             RegisterAssets(builder);
             RegisterCandles(builder);
 
             builder.Populate(_services);
         }
 
+        private void RegisterRedis(ContainerBuilder builder)
+        {
+            builder.Register(c => ConnectionMultiplexer.Connect(_redisSettings.Configuration))
```
[the remark this answers is not in the document]
Comment (dmitrysmall) [reply src/Lykke.Service.CandlesHistory/DependencyInjection/ApiModule.cs]: As far as I know, there will remain the single Redis client in bounds of the service. But probably there will be configured several instances of the service itself. In this case, each instance will get its own settings block.

File: src/Lykke.Service.CandlesHistory.Core/Domain/Candles/MarketType.cs (hunk 1 above)
Comment (dmitrysmall) [reply src/Lykke.Service.CandlesHistory.Core/Domain/Candles/MarketType.cs]: MT is a commonly-used term in Lykke. Could you explain with more details why we needed to get it renamed here?

File: src/Lykke.Service.CandlesHistory.Services/Candles/RedisCandlesCacheService.cs (hunk 4 above)
Comment (dmitrysmall) [reply src/Lykke.Service.CandlesHistory.Services/Candles/RedisCandlesCacheService.cs]: Such an initialization is still used only for purposes of an old unit test. The test was originally designed for in-memory cache service and will be fully adopted for Redis in future. Anyway, there is no a real-working scenarion with manual initializatin of the cache in CandlesHistoryService (such an initialization is to be performed in Lykke.Job.CandlesHistoryWriter startup).

File: src/Lykke.Service.CandlesHistory.Services/Candles/RedisCandlesCacheService.cs (hunk 5 above)
Comment (dmitrysmall) [reply src/Lykke.Service.CandlesHistory.Services/Candles/RedisCandlesCacheService.cs]: Such an initialization is still used only for purposes of an old unit test. Please, see the previos comment.

File: src/Lykke.Service.CandlesHistory.Core/Domain/Candles/MarketType.cs (hunk 1 above)
Comment (MikhailGerasimov) [reply src/Lykke.Service.CandlesHistory.Core/Domain/Candles/MarketType.cs]: Not everyone knows this abbreviation. But it is minor.

File: src/Lykke.Service.CandlesHistory.Services/Settings/DbSettings.cs [hunk 7]
```diff
@@ -4,6 +4,7 @@ namespace Lykke.Service.CandlesHistory.Services.Settings
 {
     public class DbSettings
     {
+        [AzureTableCheck]
         public string LogsConnectionString { get; set; }
 
         public string SnapshotsConnectionString { get; set; }
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.CandlesHistory.Services/Settings/DbSettings.cs]: Why here no check attribute?

File: src/Lykke.Service.CandlesHistory.Core/Domain/Candles/ICandlesHistoryRepository.cs [hunk 8]
```diff
@@ -7,7 +7,6 @@ namespace Lykke.Service.CandlesHistory.Core.Domain.Candles
 {
     public interface ICandlesHistoryRepository
     {
-        Task InsertOrMergeAsync(IEnumerable<ICandle> candles, string assetPairId, CandlePriceType priceType, CandleTimeInterval timeInterval);
         Task<IEnumerable<ICandle>> GetCandlesAsync(string assetPairId, CandleTimeInterval interval, CandlePriceType priceType, DateTime @from, DateTime to);
         bool CanStoreAssetPair(string assetPairId);
         Task<ICandle> TryGetFirstCandleAsync(string assetPairId, CandleTimeInterval interval, CandlePriceType priceType);
```
Comment (KonstantinRyazantsev) [inline src/Lykke.Service.CandlesHistory.Core/Domain/Candles/ICandlesHistoryRepository.cs]: Unused method

File: src/Lykke.Service.CandlesHistory.Services/Settings/SlackNotificationsSettings.cs [hunk 9]
```diff
@@ -1,7 +1,10 @@
-﻿namespace Lykke.Service.CandlesHistory.Services.Settings
+﻿using Lykke.SettingsReader.Attributes;
+
+namespace Lykke.Service.CandlesHistory.Services.Settings
 {
     public class SlackNotificationsSettings
     {
+        [AzureQueueCheck]
```
Comment (KonstantinRyazantsev) [inline src/Lykke.Service.CandlesHistory.Services/Settings/SlackNotificationsSettings.cs]: This attribute is inteded for connections tring, not complex type

**Pull request merged**